source/server/active_tcp_socket.cc

```diff
@@ -89,14 +105,25 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
           no_error = false;
           break;
         } else {
+          // There may already have data peeked due to previous filter.
+          if ((*iter_)->inspectSize() > 0 && listener_filter_buffer_->length() > 0) {
+            Network::FilterStatus status = (*iter_)->onData(*listener_filter_buffer_);
```

Comment:
it would be nice to bring the inline read with onAccept back.
Basically, it's very likely the data is already there when the socket is accepted. The first listener filter should always poke the peek instead of waiting another poller cycle

Reply:
got it, let me add a method to ListenerFilterBuffer to enable trigger the peek here.

Reply:
I trigger the peek manually at https://github.com/envoyproxy/envoy/pull/17395/files#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R95